game_helper_clients/test_client/src/sc/TestClient.java

```diff
@@ -164,60 +177,72 @@ protected void onObject(ProtocolMessage message) {
         }
         for (int i = 0; i < 2; i++)
           players[i].proc.destroy();
-
-        if (currentTests == numberOfTests)
+        
+        if (finishedTests == totalTests)
           terminateWhenPossible = true;
-        prepareNewClients();
+        else
+          prepareNewClients();
       }
     } else if (message instanceof PlayerScorePacket) {
       if (terminateWhenPossible)
         gotLastPlayerScores++;
       Score score = ((PlayerScorePacket) message).getScore();
-
+      
       for (Player p : players) {
         if (p.displayName.equals(score.getDisplayName())) {
           p.score = score;
           break;
         }
       }
-
+      
       List<ScoreValue> val = score.getScoreValues();
-      logger.warn(String.format("New score for %s: Siegpunkte %s, \u2205Feldnummer %5.2f, \u2205Karotten %5.2f after %s of %s tests",
-              score.getDisplayName(), val.get(0).getValue(), val.get(1).getValue(), val.get(2).getValue(), currentTests, numberOfTests));
-
+      logger.info(String.format("New score for %s: Siegpunkte %s, \u2205Feldnummer %5.2f, \u2205Karotten %5.2f after %s of %s tests",
+          score.getDisplayName(), val.get(0).getValue(), val.get(1).getValue(), val.get(2).getValue(), finishedTests, totalTests));
+      
       if (gotLastPlayerScores == 2) {
-        logger.warn(String.format("End results: \n" +
-                        "=============== SCORES ================\n" +
-                        "%s: %.0f\n%s: %.0f\n" +
-                        "=======================================",
-                players[0], players[0].score.getScoreValues().get(0).getValue(),
-                players[1], players[1].score.getScoreValues().get(0).getValue()));
-        logger.warn("{} of {} games did not end regularly!", irregularGames, currentTests);
+        printScores();
         exit(0);
       }
-
+      
     } else if (message instanceof PrepareGameProtocolMessage) {
       logger.debug("Received PrepareGame - starting clients");
       PrepareGameProtocolMessage pgm = (PrepareGameProtocolMessage) message;
       send(new ObservationRequest(pgm.getRoomId()));
       try {
         for (int i = 0; i < 2; i++)
-          startPlayer(i, pgm.getReservations().get((currentTests + i) % 2));
-
-        for (int i = 0; i < 2; i++)
-          if (!players[i].proc.isAlive()) {
-            logger.error("{} could not be started, look into {}", players[0].displayName, logDir);
-            exit(2);
+          startPlayer(i, pgm.getReservations().get((finishedTests + i) % 2));
+        
+        waiter.execute(() -> {
+          int tests = finishedTests;
+          int slept = 0;
+          while (tests == finishedTests) {
+            if (slept < 10)
```

Comment:
warum `10`? Keine magic numbers!

Reply:
10 Sekunden sind genug Zeit, damit fehlerhafte Clients erkannt werden, aber in der Zeit kann das Spiel noch nicht beendet sein, was nämlich hier sonst zu falschen Fehlern führen kann. Was fändest du denn besser?